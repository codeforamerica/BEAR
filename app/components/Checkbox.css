.checkbox {
  display: inline-block !important;
  background-color: #ebfffa !important;
  padding: 0 !important;
  height: 54px;
  width: 54px;
}

.checkbox:focus-within {
  box-shadow: none !important;
}

.checkbox input[type='checkbox'] {
  top: 14px !important;
  left: 14px !important;
}
