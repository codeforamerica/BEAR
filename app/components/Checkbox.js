// @flow
import * as React from 'react';
import styles from './Checkbox.css';

type Props = {
  checked: boolean,
  children: React.Node,
  group: string,
  labelText: string,
  value: string,
  onChange: (string, string) => void
};

export default class Checkbox extends React.Component<Props> {
  handleChange = () => {
    const { onChange, group, value } = this.props;
    onChange(group, value);
  };

  render() {
    const { checked, group, labelText, children } = this.props;

    return (
      <div className="form-group grid">
        <div className="form-question grid__item width-two-thirds">
          {children}
        </div>
        <label
          htmlFor={`true_${group}`}
          className={`shift-fourth-fifths ${styles.checkbox} checkbox`}
        >
          <span className="sr-only">{labelText}</span>
          <input
            defaultChecked={checked}
            type="checkbox"
            name={group}
            value={true}
            id={`true_${group}`}
            onChange={this.handleChange}
          />
        </label>
      </div>
    );
  }
}
