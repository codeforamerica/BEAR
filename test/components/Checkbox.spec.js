import sinon from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Checkbox from '../../app/components/Checkbox';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup(checked) {
  const onChangeSpy = sandbox.spy();
  const component = shallow(
    <Checkbox
      labelText="Caterine Vauban"
      checked={checked}
      group="nihilists"
      onChange={onChangeSpy}
    >
      This is a checkbox
    </Checkbox>
  );
  return {
    component,
    onChangeSpy
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('Checkbox component', () => {
  it('wraps the input in a label with proper class', () => {
    const { component } = setup(true);
    const label = component.find('label');
    expect(label.props().className).toContain('checkbox');
    expect(label.find('input').length).toBe(1);
  });

  it('renders a checkbox with value=true', () => {
    const { component } = setup(true);
    expect(component.find('input').props().value).toEqual(true);
  });

  it('is checked when selected is true', () => {
    const { component } = setup(true);
    expect(component.find('input').props().defaultChecked).toEqual(true);
  });

  it('is not checked when selected is false', () => {
    const { component } = setup(false);
    expect(component.find('input').props().defaultChecked).toEqual(false);
  });

  it('sets the name to the group prop', () => {
    const { component } = setup(false);
    expect(component.find('input').props().name).toEqual('nihilists');
  });

  it('constructs the id from the group and value', () => {
    const { component } = setup(false);
    expect(component.find('input').props().id).toEqual('true_nihilists');
  });

  it('calls onChange with the group and value when checkbox is clicked', () => {
    const { component, onChangeSpy } = setup();
    component.find('input').simulate('change');

    expect(onChangeSpy.called).toEqual(true);
    const { args } = onChangeSpy.getCall(0);
    expect(args[0]).toEqual('nihilists');
  });

  it('should match exact snapshot', () => {
    const { component } = setup();
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
