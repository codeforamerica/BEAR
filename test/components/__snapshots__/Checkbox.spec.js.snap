// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should match exact snapshot 1`] = `
<div
  className="form-group grid"
>
  <div
    className="form-question grid__item width-two-thirds"
  >
    This is a checkbox
  </div>
  <label
    className="shift-fourth-fifths checkbox checkbox"
    htmlFor="true_nihilists"
  >
    <span
      className="sr-only"
    >
      Caterine Vauban
    </span>
    <input
      id="true_nihilists"
      name="nihilists"
      onChange={[Function]}
      type="checkbox"
      value={true}
    />
  </label>
</div>
`;
